### PR TITLE
axmldec: init at 1.2.0

### PR DESCRIPTION
**Pull request by franciscod**
## Description of changes

Hello,

This adds https://github.com/ytsutano/axmldec. It's my first PR here, please let me know of any improvements I can make.

Thanks in advance!

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/by-name/ax/axmldec/package.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  meta = with lib; {
+    description = "Stand-alone binary AndroidManifest.xml decoder";
+    longDescription = ''
+      This tool accepts either a binary or a text XML file and prints the
+      decoded XML to the standard output or a file. It also allows you to
+      extract the decoded AndroidManifest.xml directly from an APK file.
+    '';
+    homepage = "https://github.com/ytsutano/axmldec";
+    license = licenses.isc;
+    maintainers = with maintainers; [ franciscod ];
```
Comment (AndersonTorres) [inline pkgs/by-name/ax/axmldec/package.nix:27]: You are not a maintainer yet.

Add an entry on maintainers.nix file.

File: pkgs/by-name/ax/axmldec/package.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
+{ lib, stdenv, fetchFromGitHub, cmake, boost, doxygen, zlib }:
+
+stdenv.mkDerivation rec {
+  pname = "axmldec";
+  version = "v1.2.0";
```
Comment (AndersonTorres) [inline pkgs/by-name/ax/axmldec/package.nix:5]:
```suggestion
  version = "1.2.0";
```

File: pkgs/by-name/ax/axmldec/package.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+  src = fetchFromGitHub {
+    repo = "axmldec";
+    owner = "ytsutano";
+    rev = version;
```
Comment (AndersonTorres) [inline pkgs/by-name/ax/axmldec/package.nix:10]:
```suggestion
    rev = "v${version}";
```

File: pkgs/by-name/ax/axmldec/package.nix [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    fetchSubmodules = true;
+    sha256 = "sha256-LFDZZbRDa8mQmglgS4DA/OqXp0HJZ2uqg1hbStdgvUw=";
```
Comment (AndersonTorres) [inline pkgs/by-name/ax/axmldec/package.nix:12]:
```suggestion
    hash = "sha256-LFDZZbRDa8mQmglgS4DA/OqXp0HJZ2uqg1hbStdgvUw=";
```

File: pkgs/by-name/ax/axmldec/package.nix [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  };
+
+  nativeBuildInputs = [ cmake ];
+  buildInputs = [ boost doxygen zlib ];
```
Comment (AndersonTorres) [inline pkgs/by-name/ax/axmldec/package.nix:16]: doxygen is usually a nativeBuildInput.

File: pkgs/by-name/ax/axmldec/package.nix [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+  meta = with lib; {
+    description = "Stand-alone binary AndroidManifest.xml decoder";
+    longDescription = ''
+      This tool accepts either a binary or a text XML file and prints the
+      decoded XML to the standard output or a file. It also allows you to
+      extract the decoded AndroidManifest.xml directly from an APK file.
+    '';
+    homepage = "https://github.com/ytsutano/axmldec";
+    license = licenses.isc;
```
Comment (AndersonTorres) [inline pkgs/by-name/ax/axmldec/package.nix:26]: Add a mainProgram entry.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by franciscod** [comment]
Thank you for your feedback @AndersonTorres :)

File: maintainers/maintainer-list.nix [hunk 7]
```diff
@@ -5954,6 +5954,11 @@
     githubId = 119691;
     name = "Michael Gough";
   };
+  franciscod = {
```
Comment (AndersonTorres) [inline maintainers/maintainer-list.nix:5957]: Put this in a separate commit (no need for a new PR)

File: pkgs/by-name/ax/axmldec/package.nix [hunk 8]
```diff
@@ -0,0 +1,31 @@
+{ lib, stdenv, fetchFromGitHub, cmake, doxygen, boost, zlib }:
```
Comment (AndersonTorres) [inline pkgs/by-name/ax/axmldec/package.nix:1]: reword the commit message to

`axmldec: init at 1.2.0`

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by franciscod** [comment]
Thanks again!

File: pkgs/by-name/ax/axmldec/package.nix (hunk 8 above)
Comment (h7x4) [inline pkgs/by-name/ax/axmldec/package.nix:1]: This is completely optional, but laying out the arguments with newlines make it easier for reviewers to look and comment on diffs later

```suggestion
{ lib
, stdenv
, fetchFromGitHub
, cmake
, doxygen
, boost
, zlib
}:
```

File: pkgs/by-name/ax/axmldec/package.nix [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+  nativeBuildInputs = [ cmake doxygen ];
+  buildInputs = [ boost zlib ];
+
+  meta = with lib; {
+    mainProgram = "axmldec";
+    description = "Stand-alone binary AndroidManifest.xml decoder";
+    longDescription = ''
+      This tool accepts either a binary or a text XML file and prints the
+      decoded XML to the standard output or a file. It also allows you to
+      extract the decoded AndroidManifest.xml directly from an APK file.
```
Comment (h7x4) [inline pkgs/by-name/ax/axmldec/package.nix:24]:
```suggestion
      extract the decoded {file}`AndroidManifest.xml` directly from an APK file.
```

File: pkgs/by-name/ax/axmldec/package.nix [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+    '';
+    homepage = "https://github.com/ytsutano/axmldec";
+    license = licenses.isc;
+    maintainers = with maintainers; [ franciscod ];
+    platforms = platforms.unix;
```
Comment (h7x4) [inline pkgs/by-name/ax/axmldec/package.nix:29]:
```suggestion
    platforms = platforms.unix ++ platforms.windows;
```

File: pkgs/by-name/ax/axmldec/package.nix (hunk 9 above)
Comment (franciscod) [reply pkgs/by-name/ax/axmldec/package.nix:31]: Not sure what this is doing but I guess it makes sense!

File: pkgs/by-name/ax/axmldec/package.nix (hunk 10 above)
Comment (franciscod) [reply pkgs/by-name/ax/axmldec/package.nix:36]: wait what there's nix for windows?

**Comment by franciscod** [comment]
Thanks @h7x4 !

File: pkgs/by-name/ax/axmldec/package.nix (hunk 10 above)
Comment (h7x4) [reply pkgs/by-name/ax/axmldec/package.nix]: No, not really. But we keep a long list of architectures and try to keep the packages' list of supported ones equal to what upstream specifies. Have a look at https://github.com/NixOS/nixpkgs/blob/master/lib/systems/doubles.nix if you're interested

File: pkgs/by-name/ax/axmldec/package.nix (hunk 9 above)
Comment (h7x4) [reply pkgs/by-name/ax/axmldec/package.nix]: Here, have a look at this: https://nixos.org/manual/nixpkgs/stable/#sec-contributing-markup

Comment (AndersonTorres) [reply pkgs/by-name/ax/axmldec/package.nix]: @h7x4 this markup is not meant to be employed on meta.description, but on Markdown text files.

Indeed, nothing similar to it appears on Nix files, except in `#` comments.

`rg -n --column -i --heading --no-config -e \`\\\{.\*\\\}`

File: maintainers/maintainer-list.nix [hunk 11]
```diff
@@ -5954,11 +5954,6 @@
     githubId = 119691;
     name = "Michael Gough";
   };
-  franciscod = {
```
Comment (AndersonTorres) [inline maintainers/maintainer-list.nix]: Re-add.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: maintainers/maintainer-list.nix (hunk 11 above)
Comment (franciscod) [reply maintainers/maintainer-list.nix]: gah, thanks

File: pkgs/by-name/ax/axmldec/package.nix (hunk 9 above)
Comment (franciscod) [reply pkgs/by-name/ax/axmldec/package.nix]: Huh... reverted

File: maintainers/maintainer-list.nix (hunk 7 above)
Comment (AndersonTorres) [inline maintainers/maintainer-list.nix:5957]: Reword this commit as

`maintainers: add franciscod`

**Review by AndersonTorres — CHANGES_REQUESTED** [review]
Besides, LGTM.

File: pkgs/by-name/ax/axmldec/package.nix (hunk 9 above)
Comment (h7x4) [reply pkgs/by-name/ax/axmldec/package.nix:31]: I see, sorry about that. Thanks!

**Review by AndersonTorres — APPROVED** [review]

**Comment by franciscod** [comment]
Should I do anything else?

File: pkgs/by-name/ax/axmldec/package.nix [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+, zlib
+}:
+
+stdenv.mkDerivation rec {
+  pname = "axmldec";
+  version = "1.2.0";
+
+  src = fetchFromGitHub {
+    repo = "axmldec";
+    owner = "ytsutano";
```
Comment (AndersonTorres) [inline pkgs/by-name/ax/axmldec/package.nix:16]:
```suggestion
    owner = "ytsutano";
    repo = "axmldec";
```

File: pkgs/by-name/ax/axmldec/package.nix [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+    description = "Stand-alone binary AndroidManifest.xml decoder";
+    longDescription = ''
+      This tool accepts either a binary or a text XML file and prints the
+      decoded XML to the standard output or a file. It also allows you to
+      extract the decoded AndroidManifest.xml directly from an APK file.
+    '';
+    homepage = "https://github.com/ytsutano/axmldec";
+    license = licenses.isc;
+    maintainers = with maintainers; [ franciscod ];
+    platforms = platforms.unix ++ platforms.windows;
```
Comment (AndersonTorres) [inline pkgs/by-name/ax/axmldec/package.nix:36]:
```suggestion
    description = "Stand-alone binary AndroidManifest.xml decoder";
    longDescription = ''
      This tool accepts either a binary or a text XML file and prints the
      decoded XML to the standard output or a file. It also allows you to
      extract the decoded AndroidManifest.xml directly from an APK file.
    '';
    homepage = "https://github.com/ytsutano/axmldec";
    license = licenses.isc;
    mainProgram = "axmldec";
    maintainers = with maintainers; [ franciscod ];
    platforms = platforms.unix ++ platforms.windows;
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]
Optional nits. Besides, lgtm

**Comment by franciscod** [comment]
Thanks, I think it makes sense to have the github components in that order (just like the URL), and the meta sorted by key as well.

**Review by AndersonTorres — APPROVED** [review]
Ping me +24h to merge

**Comment by franciscod** [comment]
Thanks!
